### PR TITLE
Benjaabdala/specific character functionality

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR I have implemented the functionality to show a specific character selected by clicking a card view in the recycler view (showed in AllCharactersFragment) or clicking in the menu button "Show specific character" and inserting a character id that you want to be shown.
Gif showing the feature:
specificCharacterPR
Project packages structure:
image
Test image:
test_image
Coverage image:
coverage_image

### CODE REVIEW

File: app/src/main/java/com/globant/marvelmvvm/view/fragment/SpecificCharacterFragment.kt [hunk 1]
```diff
@@ -5,24 +5,96 @@ import androidx.fragment.app.Fragment
...
+import com.globant.marvelmvvm.model.SpecificCharacterModel
 import com.globant.marvelmvvm.util.Constants.CHARACTER_ID
 import com.globant.marvelmvvm.util.Constants.EMPTY_STRING
-import kotlinx.android.synthetic.main.fragment_specific_character.specific_character_fragment_character_id
+import com.globant.marvelmvvm.util.Constants.ZERO
+import com.globant.marvelmvvm.util.Data
+import com.globant.marvelmvvm.util.Event
+import com.globant.marvelmvvm.util.Status
+import com.globant.marvelmvvm.viewmodel.SpecificCharacterViewModel
+import kotlinx.android.synthetic.main.fragment_specific_character.*
```
Comment (SartiMau) [inline app/src/main/java/com/globant/marvelmvvm/view/fragment/SpecificCharacterFragment.kt]: don't use wildcard imports pls

File: app/src/main/res/drawable/dialog_insert_character_id_background.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item>
+        <shape android:shape="rectangle">
+            <size
+                android:width="300dp"
```
Comment (SartiMau) [inline app/src/main/res/drawable/dialog_insert_character_id_background.xml:6]: did you try this in smaller devices?

File: app/src/main/res/layout/dialog_insert_character_id.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (SartiMau) [inline app/src/main/res/layout/dialog_insert_character_id.xml:2]: made a **ctrl + l** in the code us not bad 😉

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -9,9 +9,13 @@
     <dimen name="text_size_character_id">25sp</dimen>
     <dimen name="margin_card_view_item">10dp</dimen>
     <dimen name="padding_card_view_item">6dp</dimen>
-    <dimen name="thumbnail_character_min_size">400dp</dimen>
+    <dimen name="card_view_thumbnail_character_min_size">400dp</dimen>
```
Comment (SartiMau) [inline app/src/main/res/values/dimens.xml:12]: did you try this in smaller devices?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/globant/marvelmvvm/view/fragment/SpecificCharacterFragment.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/marvelmvvm/view/fragment/SpecificCharacterFragment.kt]: Done!

File: app/src/main/java/com/globant/marvelmvvm/data/service/response/CharacterResponse.kt [hunk 5]
```diff
@@ -4,5 +4,7 @@ import com.globant.marvelmvvm.util.Constants.EMPTY_STRING
 
 class CharacterResponse(
     val id: String = EMPTY_STRING,
+    val name: String = EMPTY_STRING,
+    val description: String = EMPTY_STRING,
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/marvelmvvm/data/service/response/CharacterResponse.kt:8]: could use NO_DESCRIPTION constant here

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/res/layout/dialog_insert_character_id.xml (hunk 3 above)
Comment (benjaglobant) [reply app/src/main/res/layout/dialog_insert_character_id.xml:2]: Done!

File: app/src/main/res/drawable/dialog_insert_character_id_background.xml (hunk 2 above)
Comment (benjaglobant) [reply app/src/main/res/drawable/dialog_insert_character_id_background.xml:6]: Mau, I make some changes and now it works good in a smaller device! Fixed!

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (benjaglobant) [reply app/src/main/res/values/dimens.xml:12]: Mau, I make some changes and now it works good in a smaller device! Fixed!

**Review by SartiMau — APPROVED** [review]

**Pull request merged**